### PR TITLE
Added GAImport tab for no data screen

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Added GAImport tab for no data screen, #PG-2903
Requires: https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/pull/423

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by bx80** [comment]
@AltamashShaikh This has a lot of conflicts, was meant to target 5.x-dev or does it need rebasing?

**Comment by AltamashShaikh** [comment]
> @AltamashShaikh This has a lot of conflicts, was meant to target 5.x-dev or does it need rebasing?

@bx80 My apologies, it was supposed to be merged into 4.x-dev

**Comment by AltamashShaikh** [comment]
> > @AltamashShaikh This has a lot of conflicts, was meant to target 5.x-dev or does it need rebasing?
> 
> @bx80 My apologies, it was supposed to be merged into 4.x-dev, I have corrected the base

**Comment by michalkleiner** [comment]
@AltamashShaikh any specific reason why this isn't done for 5.x-dev first?

**Review by bx80 — COMMENTED** [review]
I find the `Congratulations! You have successfully configured your Google Analytics import. ` text at the bottom of the instructions potentially confusing, it's not listed as the final step 7, so if I've only completed part of the process then seeing this may lead me to believe that I've completed it and this is a status message. But maybe that's just me? :shrug: 

Should there be a UI test for the new tab?

**Comment by AltamashShaikh** [comment]
> I find the `Congratulations! You have successfully configured your Google Analytics import. ` text at the bottom of the instructions potentially confusing, it's not listed as the final step 7, so if I've only completed part of the process then seeing this may lead me to believe that I've completed it and this is a status message. But maybe that's just me? shrug
> 
> Should there be a UI test for the new tab?

@bx80 Its standard for all the tabs we don't have it as a different step, will try to add a UI test, but that will require GA plugin to be included

**Comment by sgiehl** [comment]
The problem is that this whole no data screen was technically designed and grown into a direction that produces more and more dependencies and technical dept.
We should always try to design everything in a way that plugins can hook into stuff without the requirement to check in core if a plugin is installed or active.
Unfortunately it's meanwhile too late to do that for new stuff, as everything would need to be refactored before.
I actually already had some thoughts on that topic and started to implement a POC for 5.x, but I don't have time to finish that soon.

Anyway @AltamashShaikh If this new tab is actually part of the plugin, the UI test should also be included in the plugin.

**Comment by AltamashShaikh** [comment]
@bx80 @sgiehl I have added testcases but it requires core change to reflect - https://builds-artifacts.matomo.org/matomo-org/plugin-GoogleAnalyticsImporter/PG-2903-ga-import-no-data-screen/5485116313/EmptySite_GA_emptySiteDashboard.png

**Comment by bx80** [comment]
@AltamashShaikh I see the GA importer UI test screenshot is showing cloudflare. What are the core change required here? Could you elaborate a bit?

File: plugins/SitesManager/templates/_siteWithoutDataTabs.twig [hunk 1]
```diff
@@ -192,6 +195,12 @@
         </div>
     {% endif %}
 
+    {% if showGAImportTab %}
+        <div id="ga-import" class="col s12">
+            {{ postEvent('Template.embedGAImportNoData', ga3Used) }}
```
Comment (AltamashShaikh) [inline plugins/SitesManager/templates/_siteWithoutDataTabs.twig:200]: @bx80 The code to show the tab and the event to load the below UI is in this PR and the reason we see Cloudflare currently is due to we are setting GA and Cloudflare in tests and the code to show GA tab over Cloudflare is in `getActiveTabOnLoad` which is again in this PR
Screenshot from 2023-07-12 06-41-23

**Comment by AltamashShaikh** [comment]
> @AltamashShaikh I see the GA importer UI test screenshot is showing cloudflare. What are the core change required here? Could you elaborate a bit?

@bx80 The code to show the tab and the event to load the below UI is in this PR and the reason we see Cloudflare currently is due to we are setting GA and Cloudflare in tests and the code to show GA tab over Cloudflare is in `getActiveTabOnLoad` which is again in this PR

Screenshot from 2023-07-12 06-41-23

**Comment by bx80** [comment]
Got it, so once this is merged then https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/pull/423 will pickup the changes in 4.x-dev and that UI test will show properly :+1:

**Comment by AltamashShaikh** [comment]
> Got it, so once this is merged then [matomo-org/plugin-GoogleAnalyticsImporter#423](https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/pull/423) will pickup the changes in 4.x-dev and that UI test will show properly +1

Yes :+1:

**Review by bx80 — APPROVED** [review]
This looks good to me, the no data screen correctly detected ga3 on my test site and showed the GA import tab. Both removing ga3 from the test site and/or disabling the GAImporter plugin correctly resulted in the tab not being shown.

**Comment by sgiehl** [comment]
@AltamashShaikh This one still lacks an implementation for `5.x-dev`, right?
@bx80 please only merge stuff to 4.x-dev if there already is/was a PR for 5.x-dev or if it isn't meant to be in 5.x. Otherwise we might easily forget that...

**Comment by AltamashShaikh** [comment]
@sgiehl changes for 5.x-dev is in WIP

**Pull request merged**